virt_lightning/api.py

```diff
@@ -0,0 +1,442 @@
+        except ImportError:
+            libvirt.virEventRegisterDefaultImpl()
+        _register_aio_virt_impl.aio_virt_bindinds[loop] = True
+
+
+_register_aio_virt_impl.__dict__["aio_virt_bindinds"] = {}
+
+
+def _start_domain(hv, host, context, configuration):
+    if host["distro"] not in hv.distro_available():
+        logger.error("distro not available: %s", host["distro"])
+        logger.info(
+            "Please select on of the following distro: %s", hv.distro_available()
+        )
+        exit()
+
+    if "name" not in host:
+        host["name"] = re.sub(r"[^a-zA-Z0-9-]+", "", host["distro"])
+
+    if hv.get_domain_by_name(host["name"]):
+        logger.info("Skipping {name}, already here.".format(**host))
+        return
+
+    # Unfortunatly, i can't decode that symbol
+    # that symbol more well add to check encoding block
+    logger.info("{lightning} {name} ".format(lightning=symbols.LIGHTNING.value, **host))
+
+    user_config = {
+        "groups": host.get("groups"),
+        "memory": host.get("memory"),
+        "python_interpreter": host.get("python_interpreter"),
+        "root_password": host.get("root_password", configuration.root_password),
+        "ssh_key_file": host.get("ssh_key_file", configuration.ssh_key_file),
+        "username": host.get("username"),
+        "vcpus": host.get("vcpus"),
+        "fqdn": host.get("fqdn"),
+        "default_nic_mode": host.get("default_nic_model"),
+        "bootcmd": host.get("bootcmd"),
+    }
+    domain = hv.create_domain(name=host["name"], distro=host["distro"])
+    hv.configure_domain(domain, user_config)
+    domain.context = context
+    root_disk_path = hv.create_disk(
+        name=host["name"],
+        backing_on=host["distro"],
+        size=host.get("root_disk_size", 15),
+    )
+    domain.add_root_disk(root_disk_path)
+    networks = host.get("networks", [{}])
+    for i, network in enumerate(networks):
+        if "network" not in network:
+            network["network"] = configuration.network_name
+        if i == 0 and not network.get("ipv4"):
+            network["ipv4"] = hv.get_free_ipv4()
+        network["mac"] = hv.reuse_mac_address(
+            network["network"], host["name"], network.get("ipv4")
+        )
+        domain.attach_network(**network)
+    hv.start(domain, metadata_format=host.get("metadata_format", {}))
+    return domain
+
+
+def up(virt_lightning_yaml, configuration, context="default", **kwargs):
+    """
+    Create a list of VM
+    """
+
+    def myDomainEventAgentLifecycleCallback(
+        conn, dom, state, reason, opaque
+    ):  # noqa: N802
+        if state == 1:
+            logger.info("%s %s QEMU agent found", symbols.CUSTOMS.value, dom.name())
+
+    loop = kwargs.get("loop") or asyncio.get_event_loop()
+    _register_aio_virt_impl(loop)
+
+    conn = libvirt.open(configuration.libvirt_uri)
+    hv = vl.LibvirtHypervisor(conn)
+
+    conn.setKeepAlive(5, 3)
```

Comment:
Move 5, 3 to named constants.

Reply:
I use named parameters instead: `conn.setKeepAlive(interval=5, count=3)`